salt/reboot/init.sls

```diff
@@ -0,0 +1,42 @@
+##################################################
+# Configuration for the reboot manager
+##################################################
+
+# `max_holders` contains the maximum number of lock holders for the cluster.
+{% set max_holders = salt['pillar.get']('etcd:masters', '1')|int %}
+{% if max_holders > 1 %}
+  {% set max_holders = max_holders - 1 %}
+{% endif %}
+
+# Cleanup any previous cluster information on /opensuse.org
+opensuseorg_cleanup:
+  pkg.installed:
+    - name: etcdctl
+  cmd.run:
+    - name: etcdctl
+            rm -r /opensuse.org
+    # ignore failures for this
+    - check_cmd:
+      - /bin/true
+
+# Initialize the `mutex` key as expected by the reboot manager.
+set_max_holders_mutex:
+  pkg.installed:
+    - name: curl
+  cmd.run:
+    - name: curl -L -X PUT
+            http://127.0.0.1:2379/v2/keys/opensuse.org/rebootmgr/locks/global/mutex
+            -d value="0"
+    - require:
+      - cmd: opensuseorg_cleanup
+
+# Initialize the `data` key, which is JSON data with: the maximum number of
+# holders, and a list of current holders.
+set_max_holders_data:
+  pkg.installed:
+    - name: curl
+  cmd.run:
+    - name: >-
+        curl -L -X PUT http://127.0.0.1:2379/v2/keys/opensuse.org/rebootmgr/locks/global/data -d value='{ "max":"{{ max_holders }}", "holders":[] }'
```

Comment:
I think we can use `compareAndSwap` (https://coreos.com/etcd/docs/latest/v2/api.html) for this.